src/TranslationUi/Term/Repository.php

```diff
@@ -0,0 +1,45 @@
+<?php # -*- coding: utf-8 -*-
+/*
+ * This file is part of the MultilingualPress Yoast Seo Sync package.
+ *
+ * (c) Inpsyde GmbH
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Inpsyde\MultilingualPress\YoastSeoSync\TranslationUi\Term;
+
+use Inpsyde\MultilingualPress\TranslationUi\Term\RelationshipContext;
+
+class Repository
+{
+    const OPTION_KEY = 'wpseo_taxonomy_meta';
+
+    /**
+     * @param RelationshipContext $relationshipContext
+     * @param string $optionKey
+     * @return string
+     */
+    public function optionByContext(RelationshipContext $relationshipContext, string $optionKey): string
+    {
+        if(!$optionKey) {
+            throw new \InvalidArgumentException('Option key cannot be empty.');
+        }
+
+        $option = get_blog_option($relationshipContext->remoteSiteId(), self::OPTION_KEY);
```

Comment:
Because of the filters available in `get_blog_option` I would suggest to ensure that what you get from the call is an array. Something like `(array)(get_blog_option($relationshipContext->remoteSiteId(), self::OPTION_KEY, []) ?: [])`

Reply:
Why return an array here? consumers of this method are expecting a string:
https://github.com/inpsyde/MultilingualPress-Yoast-SEO-Sync/blob/prevent-sync-yoast-metadata/src/TranslationUi/Term/Field/Title.php#L80